packages/store/addon/-private/system/references/record.ts

```diff
@@ -170,11 +170,9 @@ export default class RecordReference extends Reference {
      @return {Promise<record>} the record for this RecordReference
   */
   reload() {
-    let record = this.value();
-    if (record) {
-      return record.reload();
+    if (this._id !== null) {
+      return this.store.findRecord(this.type, this._id, { reload: true });
     }
-
-    return this.load();
+    throw new Error(`Unable to fetch record of type ${this.type} without an id`);
```

Comment:
should we assert instead of throw to avoid shipping to production? I'm not sure. Value both ways. I'm tempted to create a specialized "strictModeAssert" that we usually strip but which users can opt to leave in for chasing down bugs.